### ISSUE TITLE
How to smoothly extract packets from twRO client?

### ISSUE THREAD

**Issue by Hack3rRat** [OpenKore/openkore#3591]
fucking selfish taiwanese.

**Comment by Hack3rRat** [comment]
Btw, the errors that occur using these packet extractors are attached.

1. JCV's RO Tools v1.0.3 <--- got "error" doesn't work
![JCV's RO Tools v1 0 3](https://user-images.githubusercontent.com/106661943/175492086-c4d9caf5-1e03-40e1-a00d-217a816b1492.png)

2. Packet Length Extractor v1.1.0 <--- still "error" does not work
![Packet Length Extractor v1 1 0](https://user-images.githubusercontent.com/106661943/175492142-ff08d24e-0253-42d5-ad39-051b8b41d84a.png)

3. Packet Extractor v3 <--- Could not generate "extractor.exe"

4. peek <--- always error message...
![peek](https://user-images.githubusercontent.com/106661943/175492215-7f999c16-3fdc-4586-9f43-a878fb411242.png)

I think...is it because the client of twRO has some kind of protection?
If so, how can this problem be solved?
I need a master to help me.

**Comment by carson0321** [comment]
Yes, twRO client has protection using CheatDefender. There're many encrypted packets. `recvpacket` doesn't solve this problem. If you wanna make it botable, you need to decrypt the data using reverse engineering or hook the client to get clean packets (bypass CheatDefender).

**Comment by s936511** [comment]
packetExtractor
Use twRO client written by windhamwong and Bryan Wai.
Last updated on 12 Aug 2018.
https://github.com/OpenKore/packetExtractor

**Comment by Hack3rRat** [comment]
> packetExtractor 使用由 windhamwong 和 Bryan Wai 編寫的 twRO 客戶端。 最後更新於 2018 年 8 月 12 日 [。https://github.com/OpenKore/packetExtractor](https://github.com/OpenKore/packetExtractor)

Hey...Excuse me, Thanks for your reply.
But I have tried the link file you gave.
Although I don't really know how to use it, I tried it anyway...
First I went to download Visual Studio and use it to start "packet-extractor.sln".
It did jump out of the cmd window and asked me to open the RO client, so I did as instructed.
Here is the result:

It showed "OpenProcess failed." and asked me to close.
Also did not generate any .exe file for me to update recvpackets.txt.
So... what I want to know is, does this mean that "packetExtractor-master" also doesn't work?
Or am I doing something wrong with Visual Studio?
